smoke-test/tests/cypress/cypress/e2e/mutations/add_users.js

```diff
@@ -1,47 +1,100 @@
+  cy.enterTextInTestId("name", name);
+  cy.enterTextInTestId("password", "Example password");
+  cy.enterTextInTestId("confirmPassword", "Example password");
+  cy.mouseover("#title").click();
+  cy.waitTextVisible("Other").click();
+  cy.get("[type=submit]").click();
+  return name;
 };
 
 describe("add_user", () => {
-    it("go to user link and invite a user", () => {
-        cy.login();
-
+  it("go to user link and invite a user", () => {
+    cy.login();
+    cy.visit("/settings/identities/users");
+    cy.waitTextVisible("Invite Users");
+    cy.clickOptionWithText("Invite Users");
+    cy.waitTextVisible(/signup\?invite_token=\w{32}/)
+      .then(($elem) => {
+        const inviteLink = $elem.text();
         cy.visit("/settings/identities/users");
-        cy.waitTextVisible("Invite Users");
+        cy.logout();
+        cy.visit(inviteLink);
+        let name = tryToSignUp();
+        cy.waitTextVisible("Welcome to DataHub");
+        cy.hideOnboardingTour();
+        cy.waitTextVisible(name);
+      })
+      .then(() => {
+        cy.logout();
+        cy.visit("/signup?invite_token=bad_token");
+        tryToSignUp();
+        cy.waitTextVisible("Failed to log in! An unexpected error occurred.");
+      });
+  });
+});
 
-        cy.clickOptionWithText("Invite Users");
+describe("Reset Password", () => {
+  it("Verify you can’t generate a reset password link for a non-native user", () => {
+    cy.login();
+    cy.visit("/settings/identities/users");
+    cy.wait(2000);
+    cy.get("[data-testid=userItem-non-native]").first().click();
+    cy.get("[data-testid=resetButton]").should(($button) => {
+      const isDisabled = $button.prop("disabled");
+      if (Cypress.env("ADMIN_USERNAME") !== "datahub") {
+        expect(isDisabled).to.eq(true);
+      }
+    });
+  });
+
+  it("Generate a reset password link for a native user", () => {
+    cy.login();
+    cy.visit("/settings/identities/users");
+    cy.wait(2000);
+    cy.get("[data-testid=userItem-native]").first().click();
+    cy.get("[data-testid=resetButton]").first().click();
+    cy.get("[data-testid=refreshButton]").click();
+    cy.waitTextVisible("Generated new link to reset credentials");
+    cy.wait(2000);
 
-        cy.waitTextVisible(/signup\?invite_token=\w{32}/).then(($elem) => {
-            const inviteLink = $elem.text();
-            cy.log(inviteLink);
-            cy.visit("/settings/identities/users");
-            cy.logout();
-            cy.visit(inviteLink);
-            let name = tryToSignUp();
-            cy.waitTextVisible("Welcome to DataHub");
-            cy.hideOnboardingTour();
-            cy.waitTextVisible(name);
-        }).then(() => {
-            cy.logout();
-            cy.visit("/signup?invite_token=bad_token");
-            tryToSignUp();
-            cy.waitTextVisible("Failed to log in! An unexpected error occurred.");
-        });
+    cy.window().then((win) => {
+      cy.stub(win, "prompt");
```

Comment:
What is this for?

Reply:
When we copy link in cypress it is opening chrome window prompt to manually copy link to avoid this i have added stub to prevent prompt window to open and make to automated to copy link